editor/src/messages/tool/common_functionality/measure.rs

```diff
@@ -39,7 +39,7 @@ fn draw_zero_axis_crossings(selected_bounds: Rect, hovered_bounds: Rect, transfo
 	let min_viewport = transform.transform_point2(line_start);
 	let max_viewport = transform.transform_point2(line_end);
 
-	overlay_context.dashed_line(min_viewport, max_viewport, None, Some(8.0));
+	overlay_context.dashed_line_with_pattern(min_viewport, max_viewport, None, 2.0, 2.0);
```

Comment:
based on the function description and the input parameters the dash length and the gap length should both be set to 2px. I will look into this.

Reply:
on actual render it does seem to be 3pixels of blue and 1 pixel of blank. I have probably overlooked something in the function definition. will fix ASAP.

Reply:
Yeah, basically, I'd say it's best if you try to combine these functions into one, and then figure out how to make all existing and new usages display their intended dash spacing so the end result is a function that has parameters that behave expectedly but without changing the actual behavior of other usages. Please make note of each existing dash usage and test to ensure it looks identical afterwards compared to https://dev.graphite.rs (the latest master build). It would also be helpful if you could list those locations in a comment here so I can double check them for you.

Reply:
-  the `dashed_line` method was being used in the `grid_overlays.rs` file.

-  the `measure.rs` file ( the one currently being edited ). 

Other than that there doesn't seem to be a direct reference to the method in the entire code base. The `line` method leverages the `dashed_line` method indirectly. 
the `line` method is being directly used by the following: ( i am not yet sure if they indirectly leverage dashed_line in some manner, testing would need to be done. )

- `snapping.rs`
- `pen_tool.rs`
- `utility_functions.rs` for bezier functions.

I have refactored the original function to work properly ( as it did before ) for cases where `gap_width` needs to be set to 1.
I have also set the `gap_width` parameter to None in the appropriate places (`grid_overlays.rs` and the `line` method implementation).

Reply:
Thanks! Does this also solve the 2x2 dashes appearing as 3x1 dashes? I can do another build once you've resolved that too.

Reply:
From the things i have tested on my device so far the following seem to be working as expected: Grids, Spline Tool (dash parameters ), Line Tool( dash parameters )  and the Pen Tool ( dash parameters ). Not sure if i missed something but do look into it.

Reply:
> Thanks! Does this also solve the 2x2 dashes appearing as 3x1 dashes? I can do another build once you've resolved that too.

It seems to be an issue that was already resolved in the pre existing function with how `1` was being subtracted from the dash_width, The dash width and gap width should work properly they looked to be equidistant on my local build.